dvc/output/base.py

```diff
@@ -409,12 +409,23 @@ def get_used_cache(self, **kwargs):
             cache.external[dep.repo_pair].add(dep.def_path)
             return cache
 
-        if not self.info:
-            logger.warning(
-                "Output '{}'({}) is missing version info. Cache for it will "
-                "not be collected. Use `dvc repro` to get your pipeline up to "
-                "date.".format(self, self.stage)
+        if not self.checksum:
+            msg = (
+                "Output '{}'({}) is missing version info. "
+                "Cache for it will not be collected. "
+                "Use `dvc repro` to get your pipeline up to date.".format(
+                    self, self.stage
+                )
             )
+            if self.exists:
+                msg += (
+                    "\n"
+                    "You can also use `dvc commit {stage}` to associate "
+                    "existing '{out}' with `{stage}`.".format(
+                        out=self, stage=self.stage.relpath
```

Comment:
```suggestion
                        out=self, stage=self.stage
```
same as above :slightly_smiling_face:

Reply:
AFAIK it will result in `dvc commit 'Stage file.dvc'` ;)

Reply:
@pared Oh, but then the upper warning is broken too, right?

Reply:
I wouldn't say so: the upper message is pointing user to which stage particular output is associated to, while the lower one suggests command that can fix your situation. So in the upper one, its not crucial for `stage` to be printed just as relpath.